ariwete/pkg/config/config.go

```diff
@@ -207,7 +211,55 @@ func (c *Config) FindSetupFiles(paths []string) (*map[string]CISetup, error) {
 				return nil, err
 			}
 		}
+		err := c.ValidateCISetup(setupFile, setup)
+		if err != nil {
+			return nil, err
+		}
 		setups[path] = setup
 	}
 	return &setups, nil
 }
+
+func (c *Config) ValidateCISetup(path string, setup CISetup) error {
+	var sb strings.Builder
+	for field := range setup {
+		value, exists := c.CISetupDefaults[field]
+		if strings.HasPrefix(field, "_") {
+			// This is a comment field, no need to validate.
+			continue
+		}
+
+		// Check if the field exists.
+		if !exists {
+			sb.WriteString(fmt.Sprintf("%v: '%v' does not exist.\n", path, field))
+			sb.WriteString("\n")
+			sb.WriteString("These are the valid fields:\n")
+			keys := make([]string, 0, len(c.CISetupDefaults))
+			for k := range c.CISetupDefaults {
+				keys = append(keys, k)
+			}
+			slices.Sort(keys)
+			sb.WriteString(fmt.Sprintf("%v\n", keys))
+			break
+		}
+
+		// Check the type of the field.
+		gotType := reflect.TypeOf(value)
+		expectedType := reflect.TypeOf(setup[field])
+		if gotType != expectedType {
+			sb.WriteString(fmt.Sprintf("%v: '%v' has the wrong type.\n", path, field))
+			sb.WriteString("\n")
+			sb.WriteString(fmt.Sprintf("It's defined as '%v', but got '%v'.\n", gotType, expectedType))
```

Comment:
In the case of multiple type mismatches, it might read more like one item by removing the above \n and indenting this extra info.

Reply:
Makes sense, I'll refactor this into 1 error per package, containing all the errors for that package.